src/host.c

```diff
@@ -626,6 +626,65 @@ cache_store (const char *host, struct address_list *al)
     }
 }
 
+/* Store HOST and IP to cache. */
+bool cache_host_ip (const char *host, const char *ip)
+{
+  bool result = false;
+  const char *end = ip + strlen (ip);
+  bool is_ipv6 = is_valid_ipv6_address (ip, end);
+  bool is_ipv4 = is_valid_ipv4_address (ip, end);
+  struct address_list *al;
+#ifndef ENABLE_IPV6
+  {
```

Comment:
I would recommend to move these two branches of code under ifdef to separate functions

Reply:
Thanks, it totally makes sense!